hector_gazebo_plugins/src/gazebo_ros_magnetic.cpp

```diff
@@ -162,28 +171,60 @@ void GazeboRosMagnetic::Update()
 #if (GAZEBO_MAJOR_VERSION >= 8)
   common::Time sim_time = world->SimTime();
   double dt = updateTimer.getTimeSinceLastUpdate().Double();
-
   ignition::math::Pose3d pose = link->WorldPose();
-  ignition::math::Vector3d field = sensor_model_(pose.Rot().RotateVectorReverse(magnetic_field_world_), dt);
-
-  magnetic_field_.header.stamp = ros::Time(sim_time.sec, sim_time.nsec);
-  magnetic_field_.vector.x = field.X();
-  magnetic_field_.vector.y = field.Y();
-  magnetic_field_.vector.z = field.Z();
+  field_ = sensor_model_(pose.Rot().RotateVectorReverse(magnetic_field_world_), dt);
 #else
   common::Time sim_time = world->GetSimTime();
   double dt = updateTimer.getTimeSinceLastUpdate().Double();
 
-  math::Pose pose = link->GetWorldPose();
-  math::Vector3 field = sensor_model_(pose.rot.RotateVectorReverse(magnetic_field_world_), dt);
+  ignition::math::Pose pose = link->GetWorldPose();
```

Comment:
Are you sure that this namespace change is correct for Gazebo 7?
I would prefer it if you could keep these namespaces the same as I can not test it with gazebo 7.

Reply:
You are correct; that was my mistake.  Copy & Paste error on my end.  Fixed that.